### ISSUE TITLE
Docs: Navbar logo has extra decorative "TypeScript ESLint logo" narration

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5213]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The structure of the TypeScript ESLint logo & title in the navbar is roughly:

```html
<a href="...">
  <img alt="TypeScript ESLint logo" src="..." />
  TypeScript ESLint
</a>
```

As a result, screenreaders narrate it as roughly like:

```plaintext
TypeScript ESLint logo TypeScript ESLint
```

Discussed on stream with @BenDMyers on Twitch: https://www.twitch.tv/videos/1498962341. The logo doesn't add any semantic meaning to the page. We should consider it decorative. See https://www.w3.org/WAI/tutorials/images/decorative for more details.

### Affected URL(s)

*

**Comment by Josh-Cena** [comment]
`logo.alt` is included in the default template. In hindsight, I don't even know if it makes sense to provide an alt for any logo that has an accompanying text title (or even those without, considering logos are decorative)...

**Comment by Josh-Cena** [comment]
For example, if the logo image is broken, I don't think we should show an alt text—should we? An empty alt seems good enough.

**Comment by JoshuaKGoldberg** [comment]
Yeah if the image is decorative there's no need to include an alt.

**Comment by BenDMyers** [comment]
W3C's Web Accessibility Initiative has [guidance on alt text for functional images](https://www.w3.org/WAI/tutorials/images/functional/), which specifically calls out logos used within links.

Their [first example](https://www.w3.org/WAI/tutorials/images/functional/#example-1-image-used-alone-as-a-linked-logo) is an image used alone as a linked logo. Here, they recommend supplying alt text, since the image's alt text will be used to populate the link's accessible name for assistive tech.

Their [second example](https://www.w3.org/WAI/tutorials/images/functional/#example-2-logo-image-within-link-text) is what we have on the TypeScript ESLint site: a logo image within the link text.

> The image does not represent different functionality or convey other information than that already provided in the link text, so a null (empty) value is applied, (`alt=""`), to avoid redundancy and repetition. In effect the image is a decorative adjunct or visual cue to the link text…

**Comment by JoshuaKGoldberg** [comment]
![Chris Pratt in Parks and Rec reacting with joy and shock](https://user-images.githubusercontent.com/3335181/175320826-ac363cbc-91e8-4d90-9e27-213375b78dbd.gif)

**Comment by BenDMyers** [comment]
Following up on this — the Docusaurus team has cut some a v2.0.0-rc-1 release candidate that includes facebook/docusaurus#7659, which is the upstream fix for this issue. If we upgraded to that release candidate, we'd get the fix. However, since it is a release candidate, Docusaurus v2.0 seems imminent, and so it could probably just be simpler to wait for 2.0 proper.

@JoshuaKGoldberg Thoughts?

**Comment by JoshuaKGoldberg** [comment]
Yeah this bug isn't hard blocking anybody and there's a separate PR already for 2.0. Thanks for the update!

**Comment by BenDMyers** [comment]
Cool - since there's a PR up for it anyways, would you be willing to mark this as to be resolved by #5377?

**Comment by Josh-Cena** [comment]
Hey FYI 2.0 is hopefully scheduled tomorrow, after which I'll update that PR accordingly
